pallets/dday-protocol/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+        AlreadyInTransmission,
+        // Should be stored as transmission !
+        NotInTransmission,
+        // Should be stored as transmission !
+        IsListedForSale,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(
+            n: <T as frame_system::Config>::BlockNumber,
+        ) -> frame_support::weights::Weight {
+            for data in Transmissions::<T>::iter() {
+                let (nft_id, transmission_data) = data;
+                if n >= transmission_data.delivery_date.into() {
+                    if Self::transmits_capsule(nft_id, transmission_data).is_ok() {
+                        Self::deposit_event(Event::TransmissionCompleted);
+                    } else {
+                        return 0;
+                    }
+                }
+            }
+            1
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        // Add a new tranfer into storage
+        #[pallet::weight(T::WeightInfo::dday_transmission())]
+        pub fn dday_transmission(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            recipient: T::AccountId,
+            delivery_date: BlockNumber,
+        ) -> DispatchResult {
+            let account_id = ensure_signed(origin)?;
+            let capsule = T::Capsules::get_capsule(nft_id).ok_or(Error::<T>::UnknownCapsule)?;
+            ensure!(account_id == capsule.owner, Error::<T>::NotCapsuleOwner);
+
+            ensure!(
+                T::NFTs::is_listed_for_sale(nft_id) == Some(false),
+                Error::<T>::IsListedForSale
+            );
+            ensure!(
+                T::NFTs::is_in_transmission(nft_id) == Some(false),
+                Error::<T>::AlreadyInTransmission
+            );
+
+            Transmissions::<T>::insert(nft_id, TransmissionData::new(recipient, delivery_date));
+            T::NFTs::set_in_transmission(nft_id, true)?;
+            Self::deposit_event(Event::TransmissionStored);
+            Ok(())
+        }
+
+        // Cancel transmission
+        #[pallet::weight(T::WeightInfo::cancel())]
+        pub fn cancel(origin: OriginFor<T>, nft_id: NFTId) -> DispatchResult {
+            let account_id = ensure_signed(origin)?;
+            let capsule = T::Capsules::get_capsule(nft_id).ok_or(Error::<T>::UnknownCapsule)?;
+            ensure!(account_id == capsule.owner, Error::<T>::NotCapsuleOwner);
+
+            ensure!(
+                T::NFTs::is_in_transmission(nft_id) == Some(true),
+                Error::<T>::NotInTransmission
+            );
+
+            Transmissions::<T>::remove(nft_id);
+            T::NFTs::set_in_transmission(nft_id, false)?;
+            Self::deposit_event(Event::TransmissionCancelled);
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    #[transactional]
+    fn transmits_capsule(
+        nft_id: u32,
+        transmission_data: TransmissionData<T::AccountId>,
```

Comment:
Maybe only pass the `recipient`, instead of the whole TransmissionData struct. It is the only thing you are using